### PR TITLE
[RLlib] Tf2.x native.

### PR DESCRIPTION
**Pull request by sven1977**
This PR adds native tf2.x support to RLlib:
- If tf2.x is installed: If framework="tf2", use tf2.x as the framework (w/ eager and all tf2.x features).
- If tf1.x is installed, either use eager or graph-mode, depending on the "framework" config key ("tfe" -> use eager; "tf" -> use graph mode). Error if framework="tf2".

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by ericl — CHANGES_REQUESTED** [review]
Hey this PR is too big to review; can you close and split it up into smaller <500 line PRs?

**Comment by ericl** [comment]
I'm also a bit wary of having an implicit choice between tf/tf2, given how often we see differences between the frameworks such as memory leaks; different crashers; etc. Why not make tf2 explicit with "tf2" framework?

**Comment by sven1977** [comment]
@ericl This is ready for review again.

File: rllib/tests/test_env_with_subprocess.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def leaked_processes():
     _, tmp4 = tempfile.mkstemp("test_env_with_subprocess")
     register_env("subproc", lambda c: EnvWithSubprocess(c))
 
-    ray.init()
+    ray.init(local_mode=True)
```
Comment (ericl) [inline rllib/tests/test_env_with_subprocess.py]: Revert this? Local mode won't be testing subprocesses correctly.

**Review by ericl — APPROVED** [review]
lgtm. I removed a stray local_mode change.

File: rllib/tests/test_env_with_subprocess.py (hunk 1 above)
Comment (sven1977) [reply rllib/tests/test_env_with_subprocess.py]: Perfect, thanks!

**Pull request merged**